docs/en/common_usage/set_random_seed.md

```diff
@@ -0,0 +1,39 @@
+# How to Set Random Seed
+
+[PyTorch REPRODUCIBILITY](https://pytorch.org/docs/stable/notes/randomness.html) introduces two factors that affect the reproducibility of an experiment. One is the random number, and the other is the uncertainty of the implementation of certain operators.
+
+MMEngine provides the ability to set the random number and select a deterministic algorithm. Users can simply set the `randomness` parameter of the `Runner`, which eventually calls [set_random_seed](mmengine.runner.set_random_seed), and it has the following three fields:
+
+- seed: The random seed. If this argument is not set, a random number will be used.
+- differ_rand_seed: Whether to set different seeds for different processes by adding the `rank` (process index) to the seed.
+- deterministic: Whether to set deterministic options for the CUDNN backend.
+
+Let's take the [Get Started in 15 Minutes](../get_started/15_minutes.md) as an example to demonstrate how to set `randomness` in MMEngine.
+
+```python
+runner = Runner(
+    model=MMResNet50(),
+    work_dir='./work_dir',
+    train_dataloader=train_dataloader,
+    optim_wrapper=dict(optimizer=dict(type=SGD, lr=0.001, momentum=0.9)),
+    train_cfg=dict(by_epoch=True, max_epochs=5, val_interval=1),
+    val_dataloader=val_dataloader,
+    val_cfg=dict(),
+    val_evaluator=dict(type=Accuracy),
+    # adding randomness setting
+    randomness=dict(seed=0),
+)
+runner.train()
+```
+
+However, there may still be some differences between any two experiments, even with the random number set and the deterministic algorithms chosen. Here we briefly explain why and where these fluctuations come from.
+
+First, randomness not only exists in OpenMMLab algorithms like MMDetection. You may encounter it in most of the deep learning frameworks. The core reason is that the atomic operations in CUDA are unordered and random during parallel training.
+
+To further acknowledge this, you can refer to this PyTorch official [document](https://pytorch.org/docs/stable/notes/randomness.html), which explains the uncertainties an algorithm encounters during different training sessions in PyTorch and how to control them. One is the generation of random numbers, which can be controlled by setting the seed in Numpy and PyTorch. The other is the implementation of certain operators, which can be locked using some switches in PyTorch, such as `torch.use_deterministic_algorithms`. Currently, the OpenMMLab series algorithms provide interfaces to use the same seed and the same implementations during different training sessions in the same algorithm. This is why users can see the "Set random seed to 0, deterministic: True" in the log during the training.
```

Comment:
I think this paragraph is weird here. It's an explanation of why **random seed** and **nondeterministic algorithms** affect reproducibility, but these 2 factors have already been handled in the above example.

Mabye this paragraph should be moved before *"However, there may still be some differences between any two experiments"*?

Reply:
> I think this paragraph is weird here. It's an explanation of why **random seed** and **nondeterministic algorithms** affect reproducibility, but these 2 factors have already been handled in the above example.
> 
> Mabye this paragraph should be moved before _"However, there may still be some differences between any two experiments"_?

From the whole context, I think the original zhihu answer is a supplementary explanation to the "however" part. I'm not sure but I think it is okay to keep this format.